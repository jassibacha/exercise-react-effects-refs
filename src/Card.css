.Card {
    width: 150px;
    height: auto;
    margin: 5px;
}
.Card-image {
    max-width: 100%;
}
