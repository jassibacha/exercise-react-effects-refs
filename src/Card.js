import React from 'react';
import './Card.css';

function Card({
    id,
    value,
    suit,
    image,
    rotation,
    topOffset,
    leftOffset,
    zIndex,
}) {
    return (
        <div
            className={`Card ${id}`}
            style={{
                '--random': rotation,
                top: `${topOffset}px`,
                left: `${leftOffset}px`,
                zIndex: zIndex,
            }}
        >
            <img
                src={image}
                className="Card-image"
                alt={`Card ${value} of ${suit}`}
            />
        </div>
    );
}

export default Card;
