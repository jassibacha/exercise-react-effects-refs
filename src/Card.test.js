import React from 'react';
import { render } from '@testing-library/react';
import Card from './Card';

const props = {
    id: '0',
    value: 'Ace',
    suit: 'Hearts',
    image: 'some-url',
    rotation: 0.5,
    topOffset: 10,
    leftOffset: 10,
    zIndex: 1,
};

it('renders Card component without crashing', () => {
    render(<Card {...props} />);
});

it('matches snapshot', () => {
    const { asFragment } = render(<Card {...props} />);
    expect(asFragment()).toMatchSnapshot();
});
