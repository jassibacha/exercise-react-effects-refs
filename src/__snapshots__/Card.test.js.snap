// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card 0"
    style="--random: 0.5; top: 10px; left: 10px; z-index: 1;"
  >
    <img
      alt="Card Ace of Hearts"
      class="Card-image"
      src="some-url"
    />
  </div>
</DocumentFragment>
`;
